### PR TITLE
Added RTC

### PR DESCRIPTION
**Pull request by DrDeano**
With interrupts

Added I/O waits to PIC remapping

Added fmt step to build 

When building will format all the code to the standard

Fixed cascading interrupts

### CODE REVIEW

File: build.zig [hunk 1]
```diff
@@ -29,6 +29,13 @@ pub fn build(b: *Builder) !void {
         else => unreachable,
     };
 
+    const fmt_step = b.addFmt(&[_][]const u8{
+        "build.zig",
+        "src",
+        "test",
+    });
+    b.default_step.dependOn(&fmt_step.step);
+
```
Comment (SamTebbs33) [inline build.zig]: I feel like this should be in a separate PR.

File: src/kernel/arch/x86/cmos.zig [hunk 2]
```diff
@@ -0,0 +1,458 @@
...
+const REGISTER_CENTURY: u8 = 0x32;
+
+/// The register location for return the status A register.
+const STATUS_REGISTER_A: u8 = 0x0A;
+
+/// The register location for return the status B register.
+const STATUS_REGISTER_B: u8 = 0x0B;
+
+/// The register location for return the status C register.
+const STATUS_REGISTER_C: u8 = 0x0C;
```
Comment (SamTebbs33) [inline src/kernel/arch/x86/cmos.zig]: Could these `STATUS_REGISTER_*` values be returned from a function in the `StatusRegister` enum? Just like `CounterSelect` in pit.zig.

File: src/kernel/arch/x86/cmos.zig [hunk 3]
```diff
@@ -0,0 +1,458 @@
...
+const REGISTER_DAY: u8 = 0x07;
+
+/// The register location for returning the month, (0 - 11).
+const REGISTER_MONTH: u8 = 0x08;
+
+/// The register location for returning the year, (0 - 99).
+const REGISTER_YEAR: u8 = 0x09;
+
+/// The register location for returning the century.
+const REGISTER_CENTURY: u8 = 0x32;
```
Comment (SamTebbs33) [inline src/kernel/arch/x86/cmos.zig]: Like the status register variables, these could also be returned from a function in RtcRegister.

File: src/kernel/arch/x86/cmos.zig [hunk 4]
```diff
@@ -0,0 +1,458 @@
...
+/// has time to select the register.
+///
+/// Arguments:
+///     IN reg: u8                    - The register index to select in the CMOS chip.
+///     IN comptime disable_nmi: bool - Whether to disable NMI when selecting a register.
+///
+/// Return: u8
+///     The value in the selected register.
+///
+inline fn readSelectedRegister(reg: u8, comptime disable_nmi: bool) u8 {
```
Comment (SamTebbs33) [inline src/kernel/arch/x86/cmos.zig]: I think a better API would be to take a StatusRegister instance. Also, `readSelectedRegister` sounds as though it reads an already-selected register. Perhaps `selectAndReadRegister` would be better?

File: src/kernel/arch/x86/cmos.zig [hunk 5]
```diff
@@ -0,0 +1,458 @@
...
+///
+/// Select then write to a register to the CMOS chip. This include a I/O wait to ensure the CMOS chip
+/// has time to select the register.
+///
+/// Arguments:
+///     IN reg: u8                    - The register index to select in the CMOS chip.
+///     IN data: u8                   - The data to write to the selected register.
+///     IN comptime disable_nmi: bool - Whether to disable NMI when selecting a register.
+///
+inline fn writeSelectedRegister(reg: u8, data: u8, comptime disable_nmi: bool) void {
```
Comment (SamTebbs33) [inline src/kernel/arch/x86/cmos.zig]: See comment about `readSelectedRegister`.

File: src/kernel/arch/x86/cmos.zig [hunk 6]
```diff
@@ -0,0 +1,458 @@
...
+///
+/// Read a register that corresponds to a real time clock register.
+///
+/// Arguments:
+///     IN reg: RtcRegister - A RTC register to select in the CMOS chip.
+///
+/// Return: u8
+///     The value in the selected register.
+///
+pub fn readRtcRegister(reg: RtcRegister) u8 {
```
Comment (SamTebbs33) [inline src/kernel/arch/x86/cmos.zig]: Taking an `RtcRegister` instance in the register read functions would make this unnecessary.

File: src/kernel/arch/x86/cmos.zig [hunk 7]
```diff
@@ -0,0 +1,458 @@
...
+/// Read a status register in the CMOS chip.
+///
+/// Arguments:
+///     IN reg: StatusRegister        - The status register to select.
+///     IN comptime disable_nmi: bool - Whether to disable NMI when selecting a register.
+///
+/// Return: u8
+///     The value in the selected register.
+///
+pub fn readStatusRegister(reg: StatusRegister, comptime disable_nmi: bool) u8 {
```
Comment (SamTebbs33) [inline src/kernel/arch/x86/cmos.zig]: See comment about `readRtcRegister`.

File: src/kernel/arch/x86/cmos.zig [hunk 8]
```diff
@@ -0,0 +1,458 @@
...
+
+///
+/// Write to a status register in the CMOS chip.
+///
+/// Arguments:
+///     IN reg: StatusRegister        - The status register to select.
+///     IN data: u8                   - The data to write to the selected register.
+///     IN comptime disable_nmi: bool - Whether to disable NMI when selecting a register.
+///
+pub fn writeStatusRegister(reg: StatusRegister, data: u8, comptime disable_nmi: bool) void {
```
Comment (SamTebbs33) [inline src/kernel/arch/x86/cmos.zig]: See comment about `readRtcRegister`.

File: src/kernel/arch/x86/cmos.zig (hunk 2 above)
Comment (DrDeano) [reply src/kernel/arch/x86/cmos.zig]: Can do
